vignettes/adtr.Rmd

```diff
@@ -0,0 +1,599 @@
+---
+title: "Creating ADTR"
+output: 
+  rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Creating ADTR}
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+library(admiral)
+link <- function(text, url) {
+  return(
+    paste0(
+      "[", text, "]",
+      "(", url, ")"
+    )
+  )
+}
+dyn_link <- function(text,
+                     base_url,
+                     relative_url = "",
+                     # Change to TRUE when admiral adopts multiversion docs
+                     is_multiversion = FALSE,
+                     multiversion_default_ref = "main") {
+  url <- paste(base_url, relative_url, sep = "/")
+  if (is_multiversion) {
+    url <- paste(
+      base_url,
+      Sys.getenv("BRANCH_NAME", multiversion_default_ref),
+      relative_url,
+      sep = "/"
+    )
+  }
+  return(link(text, url))
+}
+# Other variables
+admiral_homepage <- "https://pharmaverse.github.io/admiral"
+library(admiraldev)
+```
+
+# Introduction
+
+This article describes creating an `ADTR` (Tumor Results) ADaM with common
+oncology parameters based on RECIST v1.1.
+
+The main part in programming a tumor results dataset is the calculation of the
+sum of diameters of all target lesions (lymph nodes & non-lymph nodes), the
+calculation of nadir, change in & percentage change from baseline, and the
+analysis flags that could be setup for reporting.  The tumor results data could
+be set up for investigator and/or Independent review facility (IRF)/Blinded
+Independent Central Review (BICR) data.
+
+The source dataset used will depend on each company,  this could be solely the
+TR domain or you may merge `TU` with `TR` to get additional data variables or to
+ensure that you're processing the same lesions as collected.
+
+Individual sum of diameter for each target lesion (from `TR`) is required to
+calculate the sum of all target lesions, this data could be taken directly from
+`TR` or additional parameters could be created in ADTR (or similar) depending on
+the additional processing required (e.g. imputation of dates, re-labeling of
+visits) and your company specifications.
+
+NEED TO UPDATE THIS PART ONCE TEMPLATE CODE IS IN. The majority of the functions
+used here exist from `{admiral}`, except for the `tte_sources` helper object,
+provided as an example from `{admiralonco}`.
+
+**Note**: *All examples assume CDISC SDTM and/or ADaM format as input unless
+otherwise specified.*
+
+## Required Packages
+
+The examples of this vignette require the following packages. (This is to be
+reviewed/updated once template code is written)
+
+```{r, warning=FALSE, message=FALSE}
```

Comment:
I notice there are no comments in any of the chunks, it might help with context in places to add some comments.  Like I see in other admiral vignettes, e.g.:

```
# Derive final lab visit date
ds_final_lab_visit <- ds %>%
  filter(DSDECOD == "FINAL LAB VISIT") %>%
  transmute(USUBJID, FINLABDT = convert_dtc_to_dt(DSSTDTC))
# Derive treatment variables
adsl <- dm %>%
  # Merge on final lab visit date
  derive_vars_merged(
    dataset_add = ds_final_lab_visit,
    by_vars = vars(USUBJID)
  )
```

Reply:
Usually there are no comments in the chunks because the purpose of the chunk is described in the text. I would add comments only if there is something in the code which needs to be explained but is not already explained in the text.

Reply:
Ok, then the text above the chunks needs to be updated in several places (particularly one chunk with many dplyr steps).

Reply:
I have added more comments.

Reply:
Great, thank you